### PR TITLE
Implement basic GET endpoint

### PR DESCRIPTION
**Pull request by calvinnhieu**
GET endpoint checks DyanmoDB for `CompletedReducerExecutions` to determine if result is ready. If ready, return `200` and the S3 prefix key pointing to where the request results are. If not ready, also return `200`(?) with appropriate messaging.

### CODE REVIEW

File: config/iam_policy_templates/matrix-service-api-lambda.json [hunk 1]
```diff
@@ -16,6 +16,17 @@
         "lambda:InvokeFunction"
       ],
       "Resource": "arn:aws:lambda:us-east-1:861229788715:function:dcp-matrix-service-driver-dev"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:ListBucket",
```
Comment (parthshahva) [inline config/iam_policy_templates/matrix-service-api-lambda.json]: It's a public read bucket. Why is this policy needed?

File: matrix/lambdas/api/core.py [hunk 2]
```diff
@@ -55,11 +59,31 @@ def post_matrix(body: dict):
             }, requests.codes.accepted
 
 
+# TODO: write tests
 def get_matrix(request_id: str):
+    dynamo_handler = DynamoHandler()
+    # TODO: error handling: invalid request_id
+    ready, ready = dynamo_handler.increment_table_field(DynamoTable.STATE_TABLE,
```
Comment (parthshahva) [inline matrix/lambdas/api/core.py]: Why are you incrementing the reducer executions in the get_matrix request?

File: matrix/lambdas/api/core.py [hunk 3]
```diff
@@ -55,11 +59,31 @@ def post_matrix(body: dict):
...
-            'status': "In Progress",
-            'key': "sample key",
-            'eta': "sample eta",
-            'message': "sample message",
+            'status': "In progress",
+            'key': "N/A",
+            'eta': "N/A",
+            'message': "Request {request_id} is still processing. Please try again later.",
             'links': [],
             }, requests.codes.ok
```
Comment (parthshahva) [inline matrix/lambdas/api/core.py]: is there a better http status code we can use here?

File: matrix/lambdas/api/core.py (hunk 2 above)
Comment (calvinnhieu) [reply matrix/lambdas/api/core.py]: Hacky way to read from the table copied from port over. Will rewrite

File: config/iam_policy_templates/matrix-service-api-lambda.json (hunk 1 above)
Comment (calvinnhieu) [reply config/iam_policy_templates/matrix-service-api-lambda.json]: Good call. Removing

File: matrix/lambdas/api/core.py (hunk 3 above)
Comment (calvinnhieu) [reply matrix/lambdas/api/core.py]: Possibly 303 See Other?
https://github.com/requests/requests/blob/master/requests/status_codes.py

Comment (brianraymor) [reply matrix/lambdas/api/core.py]: **Pick your adventure** There is no standard.

Even Microsoft suggests both of the common patterns:

[Azure REST API](https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design#asynchronous-operations) recommends:
* a 202 response to the POST with the monitor in the Location: header
* a 303 response to the GET when the operation is successful and the created resource URI in the Location:header

[Microsoft REST API Guidelines](https://github.com/Microsoft/api-guidelines/blob/vNext/Guidelines.md#1327-the-typical-flow-polling) recommends:
* a 202 response to the POST with the monitor in a proprietary header
* a 200 response to the GET when the operation is successful with the created resource URI in the message body (_resourceLocation_)

Comment (brianraymor) [reply matrix/lambdas/api/core.py]: @mckinsel - If we continue to embed the created Resource URI in the message body, could we use a _self-documenting_ name (based on Dave Roger's questions). If _key_ was renamed _expression-matrix_ (or _matrix-location_) wouldn't that be relatively clear?

**Comment by calvinnhieu** [comment]
Thanks for the comments @brianraymor - I agree we can just go with one and my vote is for the 303 with location header. I've reached out to @hannes-ucsc @NoopDog for input on the API response changes. I don't have an issue with removing `links` in v0 - @mckinsel ?

File: matrix/lambdas/api/core.py [hunk 4]
```diff
@@ -46,20 +53,44 @@ def post_matrix(body: dict):
...
-            'key': "sample key",
-            'eta': "sample eta",
-            'message': "sample message",
-            'links': [],
-            }, requests.codes.ok
+    try:
+        dynamo_handler = DynamoHandler()
+        job_state = dynamo_handler.get_table_item(DynamoTable.STATE_TABLE, request_id)
+    except MatrixException as ex:
+        return ConnexionResponse(status_code=ex.status,
```
[the remark this answers is not in the document]
Comment (calvinnhieu) [reply matrix/lambdas/api/core.py]: In this specific case, I'm providing a more user friendly/context specific error message than the one set on the exception. Also, connexion is validating the response schema as defined in the swagger spec

**Pull request merged**